lib/xref.h

```diff
@@ -158,17 +158,18 @@ extern const struct xref * const __stop_xref_array[1] DSO_LOCAL;
  */
 #define XREF_SETUP()                                                           \
 	static const struct xref _dummy_xref = {                               \
-			/* .xrefdata = */ NULL,                                \
-			/* .type = */ XREFT_NONE,                              \
-			/* .line = */ __LINE__,                                \
-			/* .file = */ __FILE__,                                \
-			/* .func = */ "dummy",                                 \
+		/* .xrefdata = */ NULL,                                        \
+		/* .type = */ XREFT_NONE,                                      \
+		/* .line = */ __LINE__,                                        \
+		/* .file = */ __FILE__,                                        \
+		/* .func = */ "dummy",                                         \
 	};                                                                     \
-	static const struct xref * const _dummy_xref_p                         \
-			__attribute__((xref_array_attr)) = &_dummy_xref;       \
-	static void __attribute__((used, _CONSTRUCTOR(1100)))                  \
-			_xref_init(void) {                                     \
+	static const struct xref *const _dummy_xref_p                          \
+		__attribute__((xref_array_attr)) = &_dummy_xref;               \
+	static void __attribute__((used, _CONSTRUCTOR(1100))) _xref_init(void) \
+	{                                                                      \
 		static struct xref_block _xref_block = {                       \
+			.next = NULL,                                          \
```

Comment:
nit: since this is a static struct this initialization seems unnecessary. What's the intent here?

Reply:
I was getting an odd C++ compilation warning (Werror) about constructor initialization. This seemed to fix it.

Reply:
Hm that makes sense :) Thanks for all updates.